test/spec/Control/Biegunka/Source/GitSpec.hs

```diff
@@ -0,0 +1,95 @@
+module Control.Biegunka.Source.GitSpec (spec) where
+import           Test.Hspec
+
+import           Control.Biegunka
+import           Control.Biegunka.Source.Git (git_)
+import           Control.Lens
+import           Data.Maybe                  (listToMaybe)
+import           System.FilePath             ((</>))
+import           System.IO.Silently          (silence)
+import           System.IO.Temp              (withSystemTempDirectory)
+import qualified System.Process              as P
+
+spec :: Spec
+spec = do
+  describe "git_" $ around withBiegunkaDirectory $ do
+
+    context "when local path doesn't exist" $
+
+      it "creates new directory and set branch correctly" $ \tmp -> do
+          let repo = tmp </> "biegunka-test"
+          silence $ biegunka (set runRoot tmp . set biegunkaRoot (tmp </> ".biegunka")) run $
```

Comment:
I think it's fine to limit the scope to unit testing here. Perhaps a test or two can be moved to a separate spec (`test/integration` ?) if you think it's useful to do so.